discord.go

```diff
@@ -258,3 +267,25 @@ func (s *Session) onResumed(se *Session, r *Resumed) {
 	// Start the heartbeat to keep the connection alive.
 	go s.heartbeat(s.wsConn, s.listening, r.HeartbeatInterval)
 }
+
+// onGuildCreate handles the guild creation event.
+func (s *Session) onGuildCreate(se *Session, gc *GuildCreate) {
+	if _, ok := s.loadedGuildMap[gc.ID]; ok {
+		s.loadedGuildMap[gc.ID] = true
+	}
+
+	// Iterate over the guilds that must be loaded, and check if they all have been.
+	numLoadedGuilds := 0
+	for _, g := range s.loadedGuildMap {
+		if g == true {
+			numLoadedGuilds++
+		}
+	}
+
+	// If guilds are fully lazy loaded, emit a 'GuildReady' event.
+	if len(s.loadedGuildMap) > 0 && numLoadedGuilds == len(s.loadedGuildMap) {
```

Comment:
Do this `len(s.loadedGuildMap) > 0` at the top of the this function, an early escape would be good, and this seems like a good one :)